packages/client/wallets/aa/src/utils/signer.ts

```diff
@@ -0,0 +1,64 @@
+import { CrossmintService } from "@/api";
+import {
+    Blockchain,
+    FireblocksNCWallet,
+    getBlockExplorerByBlockchain,
+    getChainIdByBlockchain,
+    getDisplayNameByBlockchain,
+    getTickerByBlockchain,
+    getTickerNameByBlockchain,
+    getUrlProviderByBlockchain,
+} from "@/blockchain";
+import { FireblocksNCWSigner, UserIdentifier, WalletConfig, Web3AuthSigner } from "@/types";
+import { parseToken } from "@/utils";
+import type { SmartAccountSigner } from "@alchemy/aa-core";
+import { CHAIN_NAMESPACES } from "@web3auth/base";
+import { EthereumPrivateKeyProvider } from "@web3auth/ethereum-provider";
+import { Web3Auth } from "@web3auth/single-factor-auth";
+import { convertEthersSignerToAccountSigner, getRPCProviderOwner } from "@zerodev/sdk";
+import { Signer } from "ethers";
+
+export async function createOwnerSigner(
+    user: UserIdentifier,
+    chain: Blockchain,
+    walletConfig: WalletConfig,
+    crossmintService: CrossmintService
```

Comment:
let's make it optional and check not null only for fireblocks ncw case

Reply:
What I wanted to say is: the `crossmintService` is only necessary for the `FIREBLOCKS_NCW`
When we use `WEB3_AUTH` we don't need it. 
So I'm suggesting to make the parameter optional in the `createOwnerSigner` signature, and check that it is not null when we are in a `FIREBLOCKS_NCW` scenario. (throw excpetion if null I guess)

Reply:
What you say Maxi makes sense, however, it's already marked as required in the proposal so I would leave this as an improvement to make it in the future ( we need to sync it with the proposal).

Also, what I realized is that if a AA Wallet is created without Fireblocks, then we don't save any data about this wallet and we can not use it at all. Even on Wallet Connect. I believe that it's clear but just realizing that, right?

Reply:
I'm not sure if I understand your question @MatiNavCrossmint. We store always the abstract wallet info:


await this.crossmintService.storeAbstractWallet({
                userEmail: user.email!,
                type: ZERO_DEV_TYPE,
                smartContractWalletAddress: abstractAddress,
                eoaAddress: address,
                sessionKeySignerAddress,
                version: CURRENT_VERSION,
                baseLayer: "evm",
            });